app/models/llanfair.rb

```diff
@@ -0,0 +1,113 @@
+module Llanfair
+  class Run < Run
+    def self.sti_name
+      :llanfair
+    end
+  end
+
+  class Split < Split
+  end
+
+  class Parser < BabelBridge::Parser
+    def parse(character_array)
+      # Convert the array of character from string form to an actual array
+      # Then pack the characters back to the original bytes
+      file = RunFile.pack_binary(character_array)
+      run = {}
+      run[:time] = 0
+      # A port of Nitrofski's import llanfair files function for wsplit
+      file = StringIO.new(file)
+      file.seek(197, 1)
+      goal_len = file.read(2).unpack("n")[0]
+      run[:goal] = file.read(goal_len)
```

Comment:
Title is typically game + category, and goal is usually a time, which is auto set to be your pb time if you do not change it.  Llanfair does allow you to set it to whatever you want, so you could set it to be something like "Beat X's time" if you wanted.  I wasn't completely sure if it was a good idea to include that in the name because it could just be a time.

Llanfair has some weird quirks to it that make setting a name hard since you have so few input areas to work with.